code/modules/reagents/dispenser/dispenser2.dm

```diff
@@ -6,6 +6,10 @@
 	clicksound = "button"
 	clickvol = 20
 
+	construct_state = /decl/machine_construction/default/panel_closed
+	uncreated_component_parts = null
+	stat_immune = 0
+	base_type = /obj/machinery/chemical_dispenser
```

Comment:
This potentially needs to be reset on subtypes (though see my earlier comment): basically, base_type must match what the circuitboard you want can build.